### PR TITLE
Add the definition of get_vlan_interfaces on SwitchBase

### PR DESCRIPTION
**Pull request by fbouliane**
Also ensure that the switch is exposed to the api and that the remote switch is able to call it.

### CODE REVIEW

File: netman/api/doc_config/api_samples/get_switch_hostname_vlans_vlan_interfaces.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[3000, 3001, 3002]
```
Comment (JoProvost) [inline netman/api/doc_config/api_samples/get_switch_hostname_vlans_vlan_interfaces.json]: Based on pull request #96 this is supposed to be a list of strings, the interfaces names, not a list of integers.

File: tests/adapters/switches/remote_test.py [hunk 2]
```diff
@@ -548,6 +548,38 @@ def test_get_vlans(self):
...
+                headers=self.headers
+        ).and_return(
+            Reply(
+                content=open_fixture('get_switch_hostname_vlans_vlan_interfaces.json').read(),
+                status_code=200)
+        )
+
+        interfaces = self.switch.get_vlan_interfaces(1)
+
+        assert_that(interfaces, is_([3000, 3001, 3002]))
```
Comment (JoProvost) [inline tests/adapters/switches/remote_test.py]: Based on pull request #96 this is supposed to be a list of strings, the interfaces names, not a list of integers.

File: tests/api/switch_api_test.py [hunk 3]
```diff
@@ -106,6 +106,30 @@ def test_get_vlans_can_be_empty(self):
         assert_that(code, equal_to(200))
         assert_that(result, equal_to([]))
 
+    def test_get_vlan_interfaces(self):
+        self.switch_factory.should_receive('get_switch').with_args('my.switch').and_return(self.switch_mock).once().ordered()
+        self.switch_mock.should_receive('connect').once().ordered()
+        self.switch_mock.should_receive('get_vlan_interfaces').with_args(1).and_return(
+            [3000, 3001, 3002]
```
Comment (JoProvost) [inline tests/api/switch_api_test.py]: same

**Comment by fbouliane** [comment]
Comments fixed.

**Comment by JoProvost** [comment]
:+1:

**Comment by idjaw** [comment]
LGTM

**Pull request merged**